app/models/spree/product.rb

```diff
@@ -48,7 +48,27 @@ class Product < ApplicationRecord
     validate :validate_image
     validates :price, numericality: { greater_than_or_equal_to: 0, if: ->{ new_record? } }
 
-    accepts_nested_attributes_for :variants, allow_destroy: true
+    # These validators are used to make sure the standard variant created via
+    # `ensure_standard_variant` will be valid. The are only used when creating a new product
+    with_options on: :create_and_create_standard_variant do
```

Comment:
Instead of creating a specific context name, can we just use Rails' `create` context which is automatically applied?

Reply:
We could, but I would argue, in this scenario, it's better to be explicit with a specific context.